packages/testing/src/index.ts

```diff
@@ -9,35 +9,62 @@ import { ChildProcess, spawn, StdioOptions } from 'child_process';
 import events from 'events';
 import { kill, waitOnChild } from './child-process';
 import type getPortType from 'get-port';
+import { Connection } from './test-service-client';
 
 const TEST_SERVER_EXECUTABLE_NAME = os.platform() === 'win32' ? 'test-server.exe' : 'test-server';
 
+export const DEFAULT_TEST_SERVER_PATH = path.join(__dirname, `../${TEST_SERVER_EXECUTABLE_NAME}`);
+
 /**
- * Options for {@link TestWorkflowEnvironment.create}
+ * Convenience workflow interceptors
+ *
+ * Contains a single interceptor for transforming `AssertionError`s into non
+ * retryable `ApplicationFailure`s.
  */
-export interface TestWorkflowEnvironmentOptions {
+export const workflowInterceptorModules = [path.join(__dirname, 'assert-to-failure-interceptor')];
+
+export interface TestServerSpawnerOptions {
+  /**
+   * @default {@link DEFAULT_TEST_SERVER_PATH}
+   */
+  path?: string;
   /**
-   * If `testServerSpawner` is not provided, use this value for child process stdio
+   * @default ignore
    */
-  testServerStdio?: StdioOptions;
-  testServerSpawner?(port: number): ChildProcess;
+  stdio?: StdioOptions;
+}
+
+/**
+ * A generic callback that returns a child process
+ */
+export type TestServerSpawner = (port: number) => ChildProcess;
+
+/**
+ * Options for {@link TestWorkflowEnvironment.create}
+ */
+export interface TestWorkflowEnvironmentOptions {
+  testServer?: TestServerSpawner | TestServerSpawnerOptions;
   logger?: Logger;
 }
 
+interface TestWorkflowEnvironmentOptionsWithDefaults {
+  testServerSpawner: TestServerSpawner;
+  logger: Logger;
+}
+
 function addDefaults({
-  testServerStdio = 'ignore',
-  testServerSpawner,
+  testServer,
   logger,
-}: TestWorkflowEnvironmentOptions): Required<TestWorkflowEnvironmentOptions> {
+}: TestWorkflowEnvironmentOptions): TestWorkflowEnvironmentOptionsWithDefaults {
   return {
     testServerSpawner:
-      testServerSpawner ??
-      ((port: number) =>
-        spawn(path.join(__dirname, `../${TEST_SERVER_EXECUTABLE_NAME}`), [`${port}`], {
-          stdio: testServerStdio,
-        })),
+      typeof testServer === 'function'
+        ? testServer
+        : (port: number) =>
+            spawn(testServer?.path || DEFAULT_TEST_SERVER_PATH, [`${port}`, '--enable-time-skipping'], {
```

Comment:
I wonder if test server process management and grpc connection and such should be in core instead